optaplanner-examples/src/main/java/org/optaplanner/examples/taskassigning/solver/score/TaskAssigningConstraintProvider.java

```diff
@@ -0,0 +1,80 @@
+package org.optaplanner.examples.taskassigning.solver.score;
+
+import org.optaplanner.core.api.score.buildin.bendable.BendableScore;
+import org.optaplanner.core.api.score.stream.Constraint;
+import org.optaplanner.core.api.score.stream.ConstraintFactory;
+import org.optaplanner.core.api.score.stream.ConstraintProvider;
+import org.optaplanner.core.api.score.stream.uni.UniConstraintStream;
+import org.optaplanner.examples.taskassigning.domain.Priority;
+import org.optaplanner.examples.taskassigning.domain.Task;
+
+public final class TaskAssigningConstraintProvider implements ConstraintProvider {
+
+    private static final int BENDABLE_SCORE_HARD_LEVELS_SIZE = 1;
+    private static final int BENDABLE_SCORE_SOFT_LEVELS_SIZE = 4;
+
+    private static UniConstraintStream<Task> getInitializedTaskUniStream(ConstraintFactory constraintFactory) {
+        return constraintFactory.from(Task.class)
+                .filter(task -> task.getEmployee() != null);
+    }
+
+    private static Constraint noMissingSkills(ConstraintFactory constraintFactory) {
+        return getInitializedTaskUniStream(constraintFactory)
+                .filter(task -> task.getMissingSkillCount() > 0)
+                .penalize("No missing skills",
+                        BendableScore.ofHard(BENDABLE_SCORE_HARD_LEVELS_SIZE, BENDABLE_SCORE_SOFT_LEVELS_SIZE, 0, 1),
+                        Task::getMissingSkillCount);
+    }
+
+    private static Constraint criticalPriorityBasedTaskEndTime(ConstraintFactory constraintFactory) {
+        return getInitializedTaskUniStream(constraintFactory)
+                .filter(task -> task.getPriority() == Priority.CRITICAL)
+                .penalize("Critical priority task end time",
+                        BendableScore.ofSoft(BENDABLE_SCORE_HARD_LEVELS_SIZE, BENDABLE_SCORE_SOFT_LEVELS_SIZE, 0, 1),
+                        Task::getEndTime);
+    }
+
+    private static Constraint minimizeMakespan(ConstraintFactory constraintFactory) {
+        return getInitializedTaskUniStream(constraintFactory)
+                .filter(task -> task.getNextTask() == null)
+                .penalize("Minimize makespan, latest ending employee first",
+                        BendableScore.ofSoft(BENDABLE_SCORE_HARD_LEVELS_SIZE, BENDABLE_SCORE_SOFT_LEVELS_SIZE, 1, 1),
+                        task -> task.getEndTime() * task.getEndTime());
+        /*
+         * TODO ^^^ Some constraints need exponential weighting. perhaps we should have dynamic
+         *  constraint weights, inferred from the entity?
```

Comment:
besides .penalize(), we 'll also need to support .penalizeLoadBalanceFairness() or something like that.
The squared technique isn't even watertight.
See https://www.optaplanner.org/blog/2017/02/03/FormulaForMeasuringUnfairness.html and tennis example.

Reply:
@triceo It's really hard for people to understand that blog post, but they do need exactly that. So if we build a .penalizeUnfairness() or something like that - that follows the conclusions of that blog post - than we take that pain instead of them.